### PR TITLE
feat(auth): add Magento GraphQL driver

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/src/drivers/magento/login.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  login(loginInfo: DaffLoginInfo): Observable<DaffAuthToken> {
+    return this.apollo.mutate<GenerateTokenResponse>(
+      this.queryManager.generateATokenMutation(loginInfo)
+    ).pipe(
+      unwrapResult,
+      map(data => this.transformer.transform(data))
+    )
+  }
+}
```
Comment (lderrickable) [inline libs/auth/src/drivers/magento/login.service.ts]: I think this should be generic. It is sort of halfway to being generic with the `@Inject(DaffAuthTransformer) public transformer: DaffAuthTransformerInterface<DaffAuthToken>`. But there's not really a point in injecting the transformer service with injector tokens if this whole service isn't generic. If the app dev wants to extend the `DaffLoginInfo` model, then they will need to override the transformer AND this service. If this is generic, they'll only need to override the transformer. So like:
```
export class DaffMagentoLoginService implements DaffLoginServiceInterface<
   TLoginInfo extends DaffLoginInfo, 
   TAuthToken extends DaffAuthToken,
   TCustomerRegistration extends DaffCustomerRegistration,
   TAccountRegistration extends DaffAccountRegistration
> {
  constructor(
    private apollo: Apollo,
    @Inject(DaffAuthQueryManager) public queryManager: DaffAuthQueryManagerInterface<
      TAccountRegistration<TCustomerRegistration>,
      TCustomerRegistration,
      TLoginInfo
    >,
    @Inject(DaffAuthTransformer) public transformer: DaffAuthTransformerInterface<TAuthToken>
  ) {}

  login(loginInfo: DaffLoginInfo): Observable<TAuthToken> {
    return this.apollo.mutate<GenerateTokenResponse>(
      this.queryManager.generateATokenMutation(loginInfo)
    ).pipe(
      unwrapResult,
      map(data => this.transformer.transform(data))
    )
  }
}
```
Not sure if I got everything right in that, but something like that

File: libs/auth/src/drivers/magento/register.service.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { DaffLoginServiceInterface } from '../interfaces/login-service.interface';
+import { DaffLoginInfo } from '../../models/login-info';
+import { DaffAuthToken } from '../../models/auth-token';
+import { DaffCustomerRegistration } from '../../models/customer-registration';
+import { DaffAccountRegistration } from '../../models/account-registration';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoRegisterService implements DaffRegisterServiceInterface<
```
Comment (lderrickable) [inline libs/auth/src/drivers/magento/register.service.ts]: This should be generic too

File: libs/auth/src/drivers/magento/register.service.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  register(registration: DaffAccountRegistration<DaffCustomerRegistration>): Observable<DaffAuthToken> {
+    return this.apollo.mutate(
+      this.queryManager.createACustomerMutation(registration)
+    ).pipe(
+      mergeMap(() =>
+        this.loginDriver.login({
+          email: registration.customer.email,
+          password: registration.password
+        })
```
Comment (lderrickable) [inline libs/auth/src/drivers/magento/register.service.ts]: which means this should go in a transformer

File: libs/auth/src/drivers/magento/login.service.ts (hunk 1 above)
Comment (griest024) [reply libs/auth/src/drivers/magento/login.service.ts]: Very good point. Making those changes now.

File: libs/auth/src/drivers/magento/register.service.ts (hunk 2 above)
Comment (griest024) [reply libs/auth/src/drivers/magento/register.service.ts]: 👌

File: libs/auth/src/drivers/magento/register.service.ts (hunk 3 above)
Comment (griest024) [reply libs/auth/src/drivers/magento/register.service.ts]: the argument should go into a transformer? So we need a `TAccountRegistration -> TLoginInfo` transformer?

**Pull request closed without merging**